### PR TITLE
feat: Add PHP version module

### PR DESCRIPTION
**Pull request by REBELinBLUE**
#### Description
Adds the PHP version to the prompt in directories to `.php` files or `composer.json`

#### Motivation and Context
Since Node, Ruby, Rust, Python and Go are already included

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
Screenshot 2019-08-26 at 22 52 23

#### How Has This Been Tested?
- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

This is the first time I have done anything with Rust, I based it on the Ruby module. I am unsure about the actual tests other as most of the existing Ruby ones appear to be ignored? Also not sure if I need to add anything to `tests/Dockerfile`?

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/php.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+fn get_php_version() -> Option<String> {
+    match Command::new("php").arg("-v").output() {
+        Ok(output) => Some(String::from_utf8(output.stdout).unwrap()),
+        Err(_) => None,
+    }
+}
+
+fn format_php_version(php_version: &str) -> Option<String> {
```
[the remark this answers is not in the document]
Comment (REBELinBLUE) [reply src/modules/php.rs]: Yeah just noticed this, looking into it now

File: src/module.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ pub const ALL_MODULES: &[&str] = &[
     "python",
     "ruby",
     "rust",
+    "php",
```
[the remark this answers is not in the document]
Comment (REBELinBLUE) [reply src/module.rs]: Ah! I thought there was probably somewhere else it was needed but I couldn't find it

File: src/modules/php.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (REBELinBLUE) [reply src/modules/php.rs]: Sadly not

```$ php -r 'echo PHP_VERSION;'
7.1.31-1+ubuntu16.04.1+deb.sury.org+1z``

I did a fix last night though, looks like I just forgot to push it

**Comment by REBELinBLUE** [comment]
PHP does have a constant with the "extra" stuff stored in, so I will use that to strip it out

```
$ php -r 'echo PHP_EXTRA_VERSION;'
-1+ubuntu16.04.1+deb.sury.org+1
```

**Comment by REBELinBLUE** [comment]
Or even

```
$ php -r "echo PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION.'.'.PHP_RELEASE_VERSION;"
7.1.31
```

**Comment by REBELinBLUE** [comment]
Got it fixed, just trying to fix the Pipeline Mac build

**Comment by REBELinBLUE** [comment]
Sorry I haven't had a chance to get the CI build working (the mac build is failing to install PHP) and will be away from the computer until 16th September now so will look then

**Comment by chipbuster** [comment]
Oh boy.

@matchai It looks like PRs that haven't been rebased over the new GH Actions framework still use the Azure Pipelines. Will it be worthwhile to rebase these PRs over the new CI before we continue with them?

**Comment by REBELinBLUE** [comment]
Ah yeah i've just seen that, I shall see if I can find some time this evening to rebase and try and fix the test

**Comment by REBELinBLUE** [comment]
I am now finally back from my trip so I shall get on this again, but it'll be tomorrow evening.

@aequasi at this stage no, but I do plan to look into that later on

**Comment by REBELinBLUE** [comment]
Apologizes for the delay on this, I have rebased it, fixed the issues, updated to include the config file, updated the github workflow to install PHP and updated the documentation

**Comment by REBELinBLUE** [comment]
No worries

**Comment by matchai** [comment]
@all-contributors Please add @REBELinBLUE for code, test, docs. 🎉

**Pull request merged**